librz/core/disasm.c

```diff
@@ -5469,7 +5551,21 @@ RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, i
 			}
 			ds_print_lines_right(ds);
 			ds_print_optype(ds);
-			ds_build_op_str(ds, true);
+			char *buildopstr_ret = ds_build_op_str(ds, true, markinst, markinst_num);
+
+			if (buildopstr_ret && rz_str_startswith(rz_config_get(core->config, "asm.arch"), "x86") && strcmp(rz_config_get(core->config, "asm.bits"), "32") == 0) {
```

Comment:
In general we recommend to extract long conditions into the show inline functions.